### PR TITLE
Android N: Unblock layout inflate - ANDROIDS-1671

### PR DESCRIPTION
**Pull request by andrewleonidas**
https://lumoslabs.atlassian.net/browse/ANDROIDS-1671

Rendering of the first OpenGL surface frame blocks the inflating of the native layout on Android-N, as opposed to earlier versions which initialize the native layout first. It takes several seconds due to internal initializations and immediately hits our callback that uses the rendering of the first frame to trigger the fade out of our progress animation, which has not yet been drawn.

For Android N simply skip the rending of the first frame, the OpenGL initialization still takes place but the callbacks now take place on the second frame, allowing the layout to inflate.

### CODE REVIEW

File: cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java [hunk 1]
```diff
@@ -79,26 +90,33 @@ public void onSurfaceChanged(final GL10 pGL10, final int pWidth, final int pHeig
...
 				}
-			}
 			/*
 			 * Render time MUST be counted in, or the FPS will slower than appointed.
 			*/
-			mLastTickInNanoSeconds = System.nanoTime();
-			Cocos2dxRenderer.nativeRender();
+				mLastTickInNanoSeconds = System.nanoTime();
+				Cocos2dxRenderer.nativeRender();
+			}
```
Comment (dkdoo) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: There's an unnecessary indent on this block. Can you run the code formatter on this file?

File: cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java [hunk 2]
```diff
@@ -45,10 +47,19 @@ of this software and associated documentation files (the "Software"), to deal
...
 	private int mScreenHeight;
+	private int mFrameCount;
+	private boolean mIsNOrHigher;
 
 	// ===========================================================
 	// Constructors
 	// ===========================================================
+	public Cocos2dxRenderer() {
+
+		if (android.os.Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
```
Comment (dkdoo) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: nit: `android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.N` instead? It's a bit more explicit

Comment (andrewleonidas) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: .N does not show up in the cocos2dx project, unlike lumosity, I didn't look too hard in to why, no doubt an SDK version level issue.

File: cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java [hunk 3]
```diff
@@ -79,26 +90,33 @@ public void onSurfaceChanged(final GL10 pGL10, final int pWidth, final int pHeig
 
 	@Override
 	public void onDrawFrame(final GL10 gl) {
+
+		if (mFrameCount <= ANDROID_N_FIRST_FRAME) {
+			mFrameCount++;
+		}
+
+		if (!isNOrHigher() || mFrameCount >= ANDROID_N_FIRST_FRAME) {
```
Comment (elliottetzkorn) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: nit: `if (isNOrHigher() && mFrameCount >= ANDROID_N_FIRST_FRAME) {` seems better to me. More readable at least.

Comment (elliottetzkorn) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: nit: is we're only using this method to check its negative, would it be better to make it `ifLessThanN()` and have the method reflect this? this seems a bit more readable to me too

Comment (elliottetzkorn) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: actually I see you're using this in the other open PR. nevermind..

File: cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java (hunk 1 above)
Comment (andrewleonidas) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: Are you sure? It looks ok to me. Remember the cocos2dx project does not use Lumosity formatting, it uses tabs etc.

Comment (dkdoo) [inline cocos2dx/platform/android/java/src/main/java/org/cocos2dx/lib/Cocos2dxRenderer.java]: yep, looks good once block comment positions are adjusted

**Pull request merged**